src/FastSerialization/FastSerialization.cs

```diff
@@ -153,6 +153,7 @@ interface IStreamWriter : IDisposable
 #endif
     interface IStreamReader : IDisposable
     {
+        void Read(byte[] data, int offset, int length);
```

Comment:
Can you put this physically at the end of the interface?   I would consider this an 'advanced' API and I like to the simpler more common APIs first.      

Put a doc comment on this, and also indicate that the 'length' is likely to be restricted by the internal buffer size in the particular StreamReader.   Values over 8K need closer coordination with the reader. 

I don't know where you use Read(), but I made something called 'PinnedStreamReader' that tries to solve the same problem (getting blobs from the stream) It has 'GetPointer(length)' API.     It avoids the buffer copy, which is nice.  

That was in the days before Span<byte>.   Now I would be tempted to implement 'Span<byte> GetBytes(length) which is safe.    

I understand that that is a bigger change, but we are changing an interface here, and such contracts live forever.  (we could write a Extension method that implements Read() in terms of it, so you would only need to change the implementers.  

Hmm, something to think about...